### ISSUE TITLE
Source-control gives error and removes userform from project when there's no content

### ISSUE THREAD

**Issue by fitzcee** [rubberduck-vba/Rubberduck#2706]
I'm new to rubberduckvba.  Just installed for first time, so maybe I'm missing something easy.

Source control is removing my userforms.  It is correctly handling all other modules and class modules, but deletes forms.  Any assistance is appreciated.

Steps to reproduce:
1. create new workbook
2. open vba ide
3. insert userform
4. save xlsm workbook
5. source-control -> Manage -> initialize new repository
6. pick location and click OK
7. receive error in source-control pane: "Unknown exception. Document-type component UserForm1 does not exist. Cannot import code from repository. Document-type component UserForm1 does not exist. Cannot import code from repository. Dismiss" (note: not a typo, warning sentence is repeated twice in the app)
8. Userform has been deleted from vba project

Windows 10
Excel 2010
Rubberduckvba 2.0.12.2453

RubberduckLog.txt
2017-02-17 14:56:20.2784;INFO-2.0.11.2453;Rubberduck.App;Rubberduck version 2.0.11.2453 loading:
	Operating System: Microsoft Windows NT 6.2.9200.0 x64
	Host Product: Microsoft Office 2010 x64
	Host Version: 14.0.7177.5000
	Host Executable: EXCEL.EXE;

**Comment by retailcoder** [comment]
Is there a `UserForm1.frm` file in the repository location?

**Comment by retailcoder** [comment]
Wait a minute, *document-type* isn't normal, that would be for worksheet/workbook modules...

**Comment by retailcoder** [comment]
repro log:

````
2017-02-17 18:11:43.1688;DEBUG-2.0.12.26410;Rubberduck.UI.Command.MenuItems.ParentMenus.ParentMenuItemBase;(20292006) Executing click handler for menu item '&Source Control', hash code 25013317;
2017-02-17 18:13:45.7680;TRACE-2.0.12.26410;Rubberduck.UI.SourceControl.SourceControlViewViewModel;Initializing repo;
2017-02-17 18:13:46.3201;TRACE-2.0.12.26410;Rubberduck.UI.SourceControl.SourceControlViewViewModel;Provider changed;
2017-02-17 18:13:46.3591;TRACE-2.0.12.26410;Rubberduck.UI.SourceControl.ChangesViewViewModel;Refreshing view;
2017-02-17 18:13:46.5731;TRACE-2.0.12.26410;Rubberduck.UI.SourceControl.BranchesViewViewModel;Provider changed;
2017-02-17 18:13:46.6361;TRACE-2.0.12.26410;Rubberduck.UI.SourceControl.BranchesViewViewModel;Refreshing view;
2017-02-17 18:13:46.6961;DEBUG-2.0.12.26410;Rubberduck.Parsing.VBA.RubberduckParserState;Component 'UserForm1' was removed.;
2017-02-17 18:13:46.7431;DEBUG-2.0.12.26410;Rubberduck.Parsing.VBA.ParseCoordinator;Parsing run started. (thread 10).;
2017-02-17 18:13:46.7611;DEBUG-2.0.12.26410;Rubberduck.Parsing.VBA.RubberduckParserState;Component 'UserForm1' was removed.;
2017-02-17 18:13:46.8692;TRACE-2.0.12.26410;Rubberduck.UI.SourceControl.SourceControlViewViewModel;Displaying Error with title 'Unknown exception.' and message 'Document-type component UserForm1 does not exist. Cannot import code from repository.
Document-type component UserForm1 does not exist. Cannot import code from repository.';
2017-02-17 18:13:46.8972;TRACE-2.0.12.26410;Rubberduck.UI.SourceControl.UnsyncedCommitsViewViewModel;Provider changed;
2017-02-17 18:13:46.9242;TRACE-2.0.12.26410;Rubberduck.UI.SourceControl.UnsyncedCommitsViewViewModel;Refreshing view;
2017-02-17 18:13:47.1192;DEBUG-2.0.12.26410;Rubberduck.Parsing.VBA.ParseCoordinator;Parsing run got canceled. (thread 10).;
````

Note:

 - Form apparently gets removed twice
 - Removing the form component triggers a parse task

**Comment by retailcoder** [comment]
Also `UserForm1.log`:

    Line 8: Property OleObjectBlob in UserForm1 had an invalid file reference.

The error happens because there is no code in the userform module; you can clean up your test repository by following these steps:

 - Delete the folder you selected to create the repository in
 - Edit the `SourceControl.rubberduck` configuration file under `%appdata%\Rubberduck` and remove the `<Repository>` tag for the test project.

Now add another UserForm *with some code*:

    Option Explicit

    Sub DoSomething()
    End Sub

Should do it. Even `Option Explicit` alone should be enough. Just as long as there's *something* in the code-behind.

You can now initialize the repository without any errors.

This *is* a bug indeed.

**Comment by fitzcee** [comment]
Sorry.  Still doesn't work for me.  I did eventually get it to init with userform1.  Don't remember exactly how as I tried very many things.  I think I created, let rb delete, then recreated, then initialized.  Anyway, it stuck for one round of shutdown, startup, reload.  But it deleted when I clicked the sync button in source control pane.  Thereafter, it's steady delete -- Create Userform, Sync, Userform gone.  Creating multiple forms and all get deleted at sync.  frm files are in repository though.  I did have code-behind in each.
